### PR TITLE
loss_weights as a Model attribute

### PR DESCRIPTION
**Pull request by pescap**
Passing loss_weights as a Model attribute will allow for easier access to the weights and their update during training.

### CODE REVIEW

File: deepxde/model.py [hunk 1]
```diff
@@ -118,7 +118,8 @@ def compile(
             print("Compiling model...")
         self.opt_name = optimizer
         loss_fn = losses_module.get(loss)
-        self.losshistory.set_loss_weights(loss_weights)
+        self.losshistory.set_loss_weights(loss_weights)  # Check what it is for.
```
Comment (pescap) [inline deepxde/model.py:121]: I'd be happy to understand the role of `set_loss_weights`. Is it useful? Shall we remove it?

Comment (lululxvi) [reply deepxde/model.py:121]: This is used to record the current loss weights in `self.losshistory`.

File: deepxde/model.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def compile(
             print("Compiling model...")
         self.opt_name = optimizer
         loss_fn = losses_module.get(loss)
-        self.losshistory.set_loss_weights(loss_weights)
+        self.loss_weights = loss_weights
```
Comment (lululxvi) [inline deepxde/model.py:121]: Add `self.loss_weights = None` in line 33

File: deepxde/model.py [hunk 3]
```diff
@@ -364,7 +365,7 @@ def closure():
         self.outputs_losses_test = outputs_losses_test
         self.train_step = train_step
 
-    def _compile_jax(self, lr, loss_fn, decay, loss_weights):
+    def _compile_jax(self, lr, loss_fn, decay):
```
Comment (lululxvi) [inline deepxde/model.py:368]: If loss_wieght is not None, add a error here that loss_weights is not supported.

**Pull request merged**